HealthSDK/GiniHealthSDKExample/GiniHealthSDKExample/ScreenAPICoordinator.swift

```diff
@@ -81,37 +79,99 @@ final class ScreenAPICoordinator: NSObject, Coordinator, GiniHealthTrackingDeleg
-                                self?.rootViewController.dismiss(animated: true, completion: {
-                                    self?.delegate?.presentInvoicesList(invoices: [invoice])
-                                })
-                            }
-                        case let .failure(error):
-                            GiniUtilites.Log("❌Obtaining extractions from document with id \(docId) failed with error: \(String(describing: error))", event: .error)
-                        }
-                    }
-                case .failure(let error):
-                    GiniUtilites.Log("❌ Document data fetching failed: \(String(describing: error))", event: .error)
+    private func checkIfDocumentContainsMultipleInvoices(docId: String, using healthSdk: GiniHealth) {
+        healthSdk.checkIfDocumentContainsMultipleInvoices(docId: docId) { [weak self] result in
+            switch result {
+            case .success(let multipleInvoices):
+                if !multipleInvoices {
+                    self?.fetchDocumentDataForReview(docId: docId, using: healthSdk)
+                } else {
+                    self?.presentErrorForMultipleInvoicesInDocument()
                 }
+            case .failure(let error):
+                GiniUtilites.Log("Check if document contains multiple invoices failed with: \(String(describing: error))", event: .error)
+            }
+        }
+    }
+
+    private func createHealthExtractions(from extractions: [GiniBankAPILibrary.Extraction]) -> [GiniHealthAPILibrary.Extraction] {
+        return extractions.map { extraction in
+            GiniHealthAPILibrary.Extraction(box: nil, candidates: extraction.candidates, entity: extraction.entity, value: extraction.value, name: extraction.name)
+        }
+    }
+
+    private func checkIfDocumentIsPayable(for docId: String, using healthSdk: GiniHealth, completion: @escaping (Bool) -> Void) {
+        healthSdk.checkIfDocumentIsPayable(docId: docId) { resultPayable in
+            switch resultPayable {
+            case .success(let payable):
+                completion(payable)
+            case .failure(let error):
+                GiniUtilites.Log("Check if document is payable failed with: \(String(describing: error))", event: .error)
+                completion(false)
+            }
+        }
+    }
+
+    private func fetchDocumentDataForReview(docId: String, using healthSdk: GiniHealth) {
+        healthSdk.fetchDataForReview(documentId: docId) { [weak self] resultReview in
+            switch resultReview {
+            case .success(let data):
+                self?.fetchExtractions(for: data.document, healthSdk: healthSdk, docId: docId)
+            case .failure(let error):
+                GiniUtilites.Log("Document data fetching failed: \(String(describing: error))", event: .error)
+            }
+        }
+    }
+
+    private func fetchExtractions(for document: GiniHealthSDK.Document, healthSdk: GiniHealth, docId: String) {
```

Comment:
[clarification] Do we really need here `docId`? We can take it from the document that is passed to the method already.

Reply:
No, I removed it. Thanks for notice